modules/index-service/src/test/java/org/opencastproject/index/service/impl/IndexServiceImplTest.java

```diff
@@ -900,25 +907,69 @@ public void calculateDaysChange() throws ParseException {
 
     // PST
     start = Calendar.getInstance(pst);
-    start.set(2016, 2, 25, 22, 0);
+    start.set(2016, 2, 25, 22, 0); //  March 25, 2016 2200hrs PST is a Friday (not a schedule day)
     end = Calendar.getInstance(pst);
-    end.set(2016, 2, 29, start.get(Calendar.HOUR_OF_DAY), 5);
+    end.set(2016, 2, 29, start.get(Calendar.HOUR_OF_DAY), 5); // March 29, 2016 is a Tues
     durationMillis = (end.get(Calendar.MINUTE) - start.get(Calendar.MINUTE)) * 60 * 1000;
     days = "MO,TU,WE,SA,SU"; // --> A day after when switching to UTC (22+8)
 
     periods = generatePeriods(pst, start, end, days, durationMillis);
-    assertEquals(5, periods.size());
```

Comment:
Hmm... so the original test using the same dates and timezones was wrong? Or why is the test now expecting 4 instead of 5 days?

Reply:
The original test was wrong. Here are the 5 periods returned in current r/6.x. The scheduling is to Pacific Standard TimeZone (UTC-8) in the range Friday, March 25, 2016 2200 through Tuesday March 29, 2016 2200. You can see the calculation is off. That 10AM time doesn't even work as a UTC equivalence of PST. The results include a Friday morning and Wednesday morning. [20160326T010050/20160326T010550, 20160327T010050/20160327T010550, 20160328T010050/20160328T010550, 20160329T010050/20160329T010550, 20160330T010050/20160330T010550]

These are the 4 periods returned by this pull. These periods are in the PST TZ and fall within the requested Friday, March 25, 2016 2200hrs through Tuesday March 29, 2016 2200hrs schedule request.
[20160326T220059/20160326T220559, 20160327T220059/20160327T220559, 20160328T220059/20160328T220559, 20160329T220059/20160329T220559]